src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ConflictResolution.targets

```diff
@@ -89,4 +90,49 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!--
+    _FilterDuplicateCopyLocalPublishAssets
+    Remove CopyLocalPublishAssets items that could overwrite resolved files to publish.
+    -->
+  <Target Name="_FilterDuplicateCopyLocalPublishAssets">
+    <ItemGroup>
+      <_ResolvedCopyLocalPublishAssetsWithMetadata Include="@(_ResolvedCopyLocalPublishAssets)">
+        <RelativePath>%(_ResolvedCopyLocalPublishAssets.DestinationSubDirectory)%(Filename)%(Extension)</RelativePath>
+      </_ResolvedCopyLocalPublishAssetsWithMetadata>
+    </ItemGroup>
+    
+    <CheckForDuplicateItemMetadata
+      Items="@(_ResolvedCopyLocalPublishAssetsWithMetadata);@(ResolvedFileToPublish)"
+      MetadataName="RelativePath" >
+      <Output TaskParameter="DuplicateItems" ItemName="_DuplicateResolvedCopyLocalPublishAssets" />
+    </CheckForDuplicateItemMetadata>
+
+    <ItemGroup>
+      <_ResolvedCopyLocalPublishAssets Remove="@(_ResolvedCopyLocalPublishAssets)"/>
+      <_ResolvedCopyLocalPublishAssets Include="@(_ResolvedCopyLocalPublishAssetsWithMetadata)" Exclude="@(_DuplicateResolvedCopyLocalPublishAssets)"/>
+    </ItemGroup>
+
+  </Target>
+
+  <!--
+  _HandleFileConflictsForPublish
+  Ensures there are no duplicate files being written to the publish output.
+  -->
+  <Target Name="_HandleFileConflictsForPublish"
+          AfterTargets="ComputeFilesToPublish"
+          BeforeTargets="CopyFilesToPublishDirectory;_ComputeFilesToBundle"
+          Condition="'$(ErrorOnDuplicatePublishOutputFiles)' != 'false'">
+
+    <CheckForDuplicateItemMetadata
+      Items="@(ResolvedFileToPublish)"
+      MetadataName="RelativePath"
+      ContinueOnError="false">
+      <Output TaskParameter="DuplicatesExist" PropertyName="_ResolvedFileToPublishContainsDuplicates" />
+    </CheckForDuplicateItemMetadata>
+
+    <NETSdkError Condition="'$(_ResolvedFileToPublishContainsDuplicates)' == 'true'"
+             ResourceName="DuplicatePublishOutputFiles"/>
```

Comment:
I think this error message needs to include the duplicate paths.  Otherwise it's going to be frustrating when you run into it, it doesn't give you any information to help you fix the problem.

(Unfortunately I don't think most people know about using .binlogs to investigate things like this).